File: JavaScript/7js 3/removeUrlAnchor.js
/*
|--------------------------------------------------------------------------
| Remove anchor from URL
|--------------------------------------------------------------------------
|
*/

function removeUrlAnchor(url) {
   return url.split('#')[0];
}


function removeUrlAnchor(url) {
   return url.replace(/#.*/gi, "");
}


function removeUrlAnchor(url) {
   let string = ""
   for (let i = 0; i < url.length; i++) {
      if (url[i] === "#") {
         break
      }
      string += url[i]
   }
   return string
}


function removeUrlAnchor(url) {
   return url.match(/((?!#).)+/)[0];
}


console.log(removeUrlAnchor('www.codewars.com#about'), 'www.codewars.com')
console.log(removeUrlAnchor('www.codewars.com/katas/?page=1#about'), 'www.codewars.com/katas/?page=1')
console.log(removeUrlAnchor('www.codewars.com/katas/'), 'www.codewars.com/katas/')

File: Python/7py 3/remove_url_anchor.py
"""
|--------------------------------------------------------------------------
| Remove anchor from URL
|--------------------------------------------------------------------------
|
"""

def remove_url_anchor(url):
   return url.split('#')[0]


def remove_url_anchor(url):
   return url.partition('#')[0]


def remove_url_anchor(url):
   return url[:url.index('#')] if '#' in url else url


import re
def remove_url_anchor(url):
   return re.sub('#.*', '', url)


print(remove_url_anchor("www.codewars.com#about"), "www.codewars.com")
print(remove_url_anchor("www.codewars.com/katas/?page=1#about"), "www.codewars.com/katas/?page=1")
print(remove_url_anchor("www.codewars.com/katas/"), "www.codewars.com/katas/")